src/ext.js

```diff
@@ -155,33 +173,84 @@ export default function ext({ translator, shouldHide, senseNavigation }) {
+                // bookmark and send it to the selected automation in the
+                // property panel.
                 automation: {
                   type: 'string',
                   component: 'expression-with-dropdown',
                   translation: 'Object.ActionButton.Automation',
                   ref: 'automation',
                   dropdownOnly: true,
                   options: async () => {
-                    const automations = await fetch('../api/v1/items?resourceType=automation&limit=100').then(
-                      (response) => response.json()
-                    );
-                    return automations.data.map((blend) => ({
-                      value: blend.id,
-                      label: blend.name,
+                    const automationsResponse = await fetch('../api/v1/items?resourceType=automation&limit=100');
+                    const automations = await automationsResponse.json();
+                    return automations.data.map((a) => ({
+                      value: a.id,
+                      label: a.name,
                     }));
                   },
-                  show: (data) => checkShowAction(data, 'automation'),
+                  show: (data) => checkShowAction(data, 'automation') && !multiUserAutomation,
+                },
+                automationId: {
+                  type: 'string',
+                  component: 'expression-with-dropdown',
+                  translation: 'Object.ActionButton.Automation',
+                  ref: 'automationId',
+                  dropdownOnly: false,
+                  options: async () => getAutomations(),
+                  show: (data) => checkShowAction(data, 'automation') && multiUserAutomation,
+                  change: async (data) => {
+                    const a = await getAutomation(data.automationId)
+                    data.automationShowTriggered = a.runMode === 'triggered';
+                    if(data.automationTriggered) {
+                      data.automationExecutionToken = a.executionToken
+                    }
+                    else {
+                      data.automationExecutionToken = ''
+                    }
+                  },
                 },
-                // Boolean property to instruct the automation action to create a
-                // bookmark and send it to the selected automation in the
-                // property panel.
                 automationPostData: {
                   type: 'boolean',
                   ref: 'automationPostData',
                   translation: 'Object.ActionButton.Automation.SendSelections',
                   show: (data) => checkShowAction(data, 'automation'),
                   defaultValue: false,
                 },
+                automationShowTriggered: {
+                  type: 'boolean',
+                  ref: 'automationShowTriggered',
+                  show: () => false,
```

Comment:
Why is it defined if it will be shown?

Reply:
This property is used to determine if we should show the automationTriggered option or not. The background is that the automationTriggered property should only be visible if the automation run mode === triggered. At the moment I set the automationShowTriggered property every time the user selects a new automationId - I did it this way to limit the amount of calls we need to make to the automation api.

Reply:
But you need to have it defined in the property panel? Can you not just set the value `automationShowTriggered` as you already do

Reply:
I was under the impression that I needed to have it in the props panel in order to set it but I see that is not the case :) - I will remove it